GTG/gtk/browser/browser.py

```diff
@@ -420,8 +425,76 @@ def _init_accelerators(self):
                                        Gtk.AccelFlags.VISIBLE)
 
 # HELPER FUNCTIONS ##########################################################
-    def temporary_search(self, widget):
-        self.quickadd_entry.grab_focus()
+
+    def on_search_toggled(self, widget):
+        search = self.builder.get_object("search")
+        self.bar = self.builder.get_object("search_bar")
+        self.entry = self.builder.get_object("search_entry")
+        if self.bar.get_search_mode():
+            search.set_active(False)
+            self.bar.set_search_mode(False)
+        else:
+            search.set_active(True)
+            self.bar.set_search_mode(True)
+            self.entry.grab_focus()
+
+    def on_search(self, key, widget):
+        query = self.search_entry.get_text()
+
+        try:
+            parsed_query = parse_search_query(query)
+            self.apply_filter_on_panes(SEARCH_TAG, parameters=parsed_query,
+                                       refresh=True)
+        except InvalidQuery:
+            pass
```

Comment:
:+1:  Great advice! Cheers :)
(just a note: You meant "except InvalidQuery as e:". right?)